bulldozer/evaluate.go

```diff
@@ -0,0 +1,202 @@
+
+package bulldozer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/palantir/bulldozer/pull"
+)
+
+// IsPRBlacklisted returns true if the PR is identified as blacklisted,
+// false otherwise. Additionally, a description of the reason will be returned.
+func IsPRBlacklisted(ctx context.Context, pullCtx pull.Context, config Signals) (bool, string, error) {
+	labels, err := pullCtx.Labels(ctx)
+	if err != nil {
+		return true, "unable to list PR labels", err
+	}
+
+	if inSlice, idx := anyInSlice(labels, config.Labels); inSlice {
+		return true, fmt.Sprintf("PR label matches one of specified blacklist labels: %q", config.Labels[idx]), nil
+	}
+
+	body, err := pullCtx.Body(ctx)
+	if err != nil {
+		return true, "unable to list PR body", err
+	}
+
+	comments, err := pullCtx.Comments(ctx)
+	if err != nil {
+		return true, "unable to list PR comments", err
+	}
+
+	if inSlice, idx := anyInSlice(comments, config.Comments); inSlice {
+		return true, fmt.Sprintf("PR comment matches one of specified blacklist comments: %q", config.Comments[idx]), nil
+	}
+
+	for _, blacklistedComment := range config.Comments {
+		if strings.EqualFold(blacklistedComment, body) {
+			return true, fmt.Sprintf("PR body matches one of specified blacklist comments: %q", blacklistedComment), nil
+		}
+	}
+
+	for _, blacklistedSubstring := range config.CommentSubstrings {
+		for _, comment := range comments {
+			if strings.Contains(comment, blacklistedSubstring) {
+				return true, fmt.Sprintf("PR comment matches one of specified blacklist comment substrings: %q", blacklistedSubstring), nil
+			}
+		}
+
+		if strings.Contains(body, blacklistedSubstring) {
+			return true, fmt.Sprintf("PR body matches one of specified blacklist comment substrings: %q", blacklistedSubstring), nil
+		}
+	}
+
+	return false, "no matching blacklist found", nil
+}
+
+// IsPRWhitelisted returns true if the PR is identified as whitelisted,
+// false otherwise. Additionally, a description of the reason will be returned.
+func IsPRWhitelisted(ctx context.Context, pullCtx pull.Context, config Signals) (bool, string, error) {
+	labels, err := pullCtx.Labels(ctx)
+	if err != nil {
+		return false, "unable to list PR labels", err
+	}
+
+	if inSlice, idx := anyInSlice(labels, config.Labels); inSlice {
+		return true, fmt.Sprintf("PR label matches one of specified whitelist labels: %q", config.Labels[idx]), nil
+	}
+
+	body, err := pullCtx.Body(ctx)
+	if err != nil {
+		return false, "unable to list PR body", err
+	}
+
+	comments, err := pullCtx.Comments(ctx)
+	if err != nil {
+		return false, "unable to list PR comments", err
+	}
+
+	if inSlice, idx := anyInSlice(comments, config.Comments); inSlice {
+		return true, fmt.Sprintf("PR comment matches one of specified whitelist comments: %q", config.Comments[idx]), nil
+	}
+
+	for _, whitelistedComment := range config.Comments {
+		if strings.EqualFold(whitelistedComment, body) {
```

Comment:
I believe `EqualFold` is a case-insensitive comparison. Is there a reason full comment matching is case-insensitive, but comment substring matching is case-sensitive?